Packs/MicrosoftCloudAppSecurity/Integrations/MicrosoftCloudAppSecurity/MicrosoftCloudAppSecurity.py

```diff
@@ -182,6 +226,47 @@ def list_incidents(self, filters: dict, limit: Union[int, str]):
         )
 
 
+@logger
+def start_auth(client: Client) -> CommandResults:
+    result = client.ms_client.start_auth('!microsoft-cas-auth-complete')
+    return CommandResults(readable_output=result)
+
+
+@logger
+def complete_auth(client: Client) -> CommandResults:
+    client.ms_client.get_access_token()
+    return CommandResults(readable_output='✅ Authorization completed successfully.')
+
+
+@logger
+def reset_auth() -> CommandResults:
+    set_integration_context({})
+    return CommandResults(readable_output='Authorization was reset successfully. You can now run '
+                                          '**!microsoft-cas-auth-start** and **!microsoft-cas-auth-complete**.')
+
+
+@logger
+def test_connection(client: Client) -> CommandResults:
+    test_context_for_token(client)
+    client.ms_client.get_access_token()  # If fails, MicrosoftApiModule returns an error
+    return CommandResults(readable_output='✅ Success!')
+
+
+def test_context_for_token(client: Client) -> None:
+    """test_context_for_token
+    Checks if the user acquired token via the authentication process.
+    Args:
+    Returns:
+
+    """
+    if client.client_credentials:
+        return
+    if not (get_integration_context().get('access_token') or get_integration_context().get('current_refresh_token')):
+        raise DemistoException(
+            "This integration does not have a test module. Please run !microsoft-cas-auth-auth-start and "
```

Comment:
>This integration does not have a test module

This is not true in this case. And to be honest I am not sure why it is relevant here. We need to say something like "Connection could be tested as either the access or refresh token is missing. Please run..." 
Let's sync with Yuvall on this too